File: trip-planner-front/src/app/_models/dto/locationDto.ts
import { Byte } from "@angular/compiler/src/util";

export class LocationDto {
    name: string;

    description: string;

    address: string;

    priority: string;

    duration: number;

    trivia: string;

    photo: Byte[];

    region: number;

    city: number;

    user: string;


    constructor() {
        this.name = '';
        this.description = '';
        this.address = '';
        this.priority = '';
        this.duration = 0;
        this.trivia = '';
        this.photo = [];
        this.city = 0;
        this.region = 0;
        this.user = '';
    }
}

File: trip-planner/src/main/java/finki/diplomska/tripplanner/models/dto/LocationDto.java
package finki.diplomska.tripplanner.models.dto;

import finki.diplomska.tripplanner.models.*;
import lombok.Data;

import java.util.List;

@Data
public class LocationDto {

    private String name;

    private String description;

    private String address;

    private String priority;

    private int duration;

    private String trivia;

    private byte[] photo;

    private Long region;

    private Long city;

    private String user;

    private List<Long> companions;

    private List<Long> categories;


    public LocationDto(String name, String description, String address, String priority, int duration, String trivia, byte[] photo, Long region, Long city, String user ) {
        this.name = name;
        this.description = description;
        this.address = address;
        this.priority = priority;
        this.duration = duration;
        this.trivia = trivia;
        this.photo = photo;
        this.region = region;
        this.city = city;
        this.user = user;
        this.companions = companions;
        this.categories = categories;
    }
    public LocationDto() {
    }
}
